src/ripple/shamap/SHAMap.h

```diff
@@ -182,10 +182,9 @@ class SHAMap
     
     bool getRootNode (Serializer & s, SHANodeFormat format) const;
     std::vector<uint256> getNeededHashes (int max, SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (uint256 const& hash, Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
+    SHAMapAddNode addRootNode (boost::optional<uint256> const& hash,
```

Comment:
Sorry, brain moment - I meant are you sure we should pass `const&` and not by value?